static/app/utils/profiling/renderers/textRenderer.tsx

```diff
@@ -126,19 +131,53 @@ class TextRenderer {
       // Offset x by 1x the padding
       const x = frameInPhysicalSpace[0] + HALF_SIDE_PADDING;
 
-      this.context.fillText(
-        trimTextCenter(
-          frame.frame.name,
-          findRangeBinarySearch(
-            {low: 0, high: paddedRectangleWidth},
-            n => this.measureAndCacheText(frame.frame.name.substring(0, n)),
-            paddedRectangleWidth
-          )[0]
-        ),
-        x,
-        y
+      const frameName = frame.frame.name;
+      const trim = trimTextCenter(
+        frameName,
+        findRangeBinarySearch(
+          {low: 0, high: paddedRectangleWidth},
+          n => this.measureAndCacheText(frame.frame.name.substring(0, n)).width,
+          paddedRectangleWidth
+        )[0]
       );
 
+      const {text: trimText} = trim;
+
+      if (flamegraphSearchResults) {
+        const frameId = getFlamegraphFrameSearchId(frame);
+        const frameSearchResult = flamegraphSearchResults[frameId];
+
+        if (frameSearchResult) {
+          const highlightFillStyle = `rgb(${this.theme.COLORS.HIGHLIGHTED_LABEL_COLOR.join(
+            ', '
+          )})`;
```

Comment:
I'd imagine as our theme grows and we fully transition to color channels this is going to be very repetitive, we should lean into some `utils` or a `RGB` class. it would help with this;